### ISSUE TITLE
How to ensure option is chosen?

### ISSUE THREAD

**Issue by aaroncrawford** [ericgio/react-bootstrap-typeahead#227]
Is there a trick to ensure that the user has to choose an option presented by Typeahead?  I tried changing a state variable which keeps track of the current input variable via onInputChange, then onBlur checking if the value is in the list of options.  The problem I'm running into is that when a user **clicks** the options presented, blur is (correctly) triggered, which is preventing me from utilizing that method to ensure that the final value is included in the options.

Any other workarounds that I'm not thinking of?

I tried to put this on SO with the react-bootstrap-typeahead tag, but the tag didn't exist and I didn't have enough rep to create it.

**Comment by ericgio** [comment]
I'm unclear on what you're asking here. By design, the typeahead doesn't allow selections except what is contained in the list of options unless you set the `allowNew` prop.

**Comment by aaroncrawford** [comment]
Understood - let me rephrase.  When someone begins typing into the input, options are presented.  I want to ensure they are forced to choose one of the options presented by typeahead and not free type into the input box.  They need to choose one of the options.  I tried using the blur method provided to make sure they chose one of the options, however blur is triggered when they click (not arrow key down and enter) on an option. Are there any built int methods (or string of methods) that would allow me to force them to choose an option?

**Comment by ericgio** [comment]
There's no built-in way to validate the field; that's left up to the developer and the particular requirements of the form. You can use `onChange` to get the selected value: if it's empty, then they either skipped the field or tried to enter a freeform value. You'd need to check on `submit`, however, since you're right that clicking the menu blurs the input.

**Comment by aaroncrawford** [comment]
Roger.  Thanks for your help.

**Comment by bkoltai** [comment]
I'm interested in implementing this functionality as well.

@aaroncrawford did you get it to work?

I found @leolarson's [comment](https://github.com/ericgio/react-bootstrap-typeahead/issues/224#issuecomment-326396990) that seems like a duplicate here.

What I noticed is that the `onChange` handler doesn't get called when the input is blurred. So when a user types in the input and then clicks away (not on a menu item), the `onChange` doesn't get called. When I tried to use the `onBlur`, it did successfully get called when clicking away, but since it is called before the `onChange`, I get a false positive validation error when the user is clicking on a menu item.

Would be great to see a working example of how to get this functionality. Thanks!

**Comment by aaroncrawford** [comment]
@bkoltai So I wasn't able to use any of the methods in this repo to accomplish it 100%.  I had to manually run through all of the form fields on form submit to ensure that the value of the input was in the array of options brought back by my endpoint.  All of my form fields are components so they all had their own state, which made it a bit easier for me.

I was very close to being able to pull this off with just onBlur - but as I stated earlier, blur is triggered when a user clicks on an option which made that method a nonstarter for me.
